custom_components/waste_collection_schedule/waste_collection_schedule/source/zva_wmk_de.py

```diff
@@ -0,0 +1,50 @@
+import requests
+import datetime
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+from waste_collection_schedule.service.ICS import ICS
+
+import urllib
+
+TITLE = "Zweckverband Abfallwirtschaft Werra-Meißner-Kreis"
+DESCRIPTION = "Source for Zweckverband Abfallwirtschaft Werra-Meißner-Kreis"
+URL = "https://www.zva-wmk.de/"
+TEST_CASES = {
+    "Frankenhain": {"city": "Berkatal - Frankenhain", "street": "Teichhof"},
+    "Hebenshausen": {"city": "Neu-Eichenberg - Hebenshausen", "street": "Bachstraße"},
+    "Vockerode": {"city": "Meißner - Vockerode", "street": "Feuerwehr"}
+}
+
+
+class Source:
+    def __init__(self, city, street):
+        self._city = city
+        self._street = street
+        self._ics = ICS(split_at=" / ")
+
+    def fetch(self):
+        city = self._city.upper()
+        city = self._city.replace('ß', 'ẞ').upper()
+        city = city.replace(" - ", "_")
+        city = city.replace(" ", "+")
+        city = city.replace("ẞ", "ß")
+        city = urllib.parse.quote(city.encode('utf8'))
+        street = self._street
+        street = street.replace(" ","+")
+        today = datetime.date.today()
+        year = today.year
+        if year == 2022:
+           yearstr = ""
+           street = self._street.upper()
+        else:
+           yearstr = ("-" + str(year))
+
+        urlzva = "https://www.zva-wmk.de/termine/schnellsuche"+yearstr+"&type=all&link=ical&timestart=6&fullday=1&timeend=17&reminder=1440&display=0&city="+city+"&street="+street
+
+        r = requests.get(urlzva)
+        r.encoding = r.apparent_encoding
```

Comment:
Don't use `apparent_encoding `but specify the correct encoding (typically `utf-8`) directly.

Reply:
removed